pkg/util/k8s/node.go

```diff
@@ -91,3 +101,24 @@ func IsNodeCordoned(node *v1.Node) (bool, time.Time) {
 	}
 	return false, time.Time{}
 }
+
+// GetNodeTopologyLabelKeys returns a list of sorted node label keys that exist in a candidate list
+func GetNodeTopologyLabelKeys() ([]string, error) {
+	nodeList, err := coreops.Instance().GetNodes()
```

Comment:
Let's pass the sigs controller k8s client from the caller here instead of using the sched-ops GetNodes method. The controller client will get the nodes from the cache, while the GetNodes will hit the k8s API server everytime.